src/particles/PhysicsParticles.hpp

```diff
@@ -746,19 +822,33 @@ template <typename problem_t> class PhysicsParticleRegister
 	// Update velocities of all massive particles
 	void kickParticlesAtLevel(amrex::Real dt, amrex::MultiFab &acceleration, int lev)
 	{
-		for (const auto &[name, descriptor] : particleRegistry_) {
+		for (const auto &[type, descriptor] : particleRegistry_) {
 			if (descriptor->getMassIndex() >= 0) {
 				descriptor->kickParticles(lev, dt, acceleration);
 			}
 		}
 	}
 
-	// Create CIC particles
-	void createCICParticles(amrex::MultiFab &state, int lev, amrex::Real current_time, amrex::Real dt, amrex::Real param1, amrex::Real param2)
+	// Create particles based on particle type
+	void createParticles(amrex::MultiFab &state, int lev, amrex::Real current_time, amrex::Real dt, amrex::Real param1, amrex::Real param2)
 	{
-		auto descriptor = getParticleDescriptor("CIC_particles");
-		if (descriptor != nullptr) {
-			descriptor->createCICParticles(state, lev, current_time, dt, param1, param2);
+		for (const auto &[type, descriptor] : particleRegistry_) {
+			// Only create particles if the descriptor allows creation
+			if (descriptor->getAllowsCreation()) {
+				// Dispatch to the appropriate particle type based on the particle type
+				switch (type) { // NOLINT
+					case ParticleType::CIC:
```

Comment:
I thought the templated design of the PhysicsParticleDescriptor allows us to avoid all of these kinds of explicit enumerations of particle types (except possibly when creating each descriptor).

Can we instead provide a specialization of `descriptor->createParticles` for each template type?